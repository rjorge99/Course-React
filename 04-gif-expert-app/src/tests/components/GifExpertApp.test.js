import { shallow } from 'enzyme';
import GiftExpertApp from '../../GifExpertApp';

describe('Prubas de GifExpertApp', () => {
    test('Debe mostrar el componente correctamente', () => {
        const wrapper = shallow(<GiftExpertApp />);
        expect(wrapper).toMatchSnapshot();
    });
    test('Debe mostrar una lista de categorias', () => {
        const categorias = ['1', '2'];
        const wrapper = shallow(
            <GiftExpertApp defaultCategories={categorias} />
        );

        expect(wrapper).toMatchSnapshot();
        expect(wrapper.find('GifGrid').length).toBe(categorias.length);
    });
});
