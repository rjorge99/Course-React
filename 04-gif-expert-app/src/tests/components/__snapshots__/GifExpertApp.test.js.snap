// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Prubas de GifExpertApp Debe mostrar el componente correctamente 1`] = `
<Fragment>
  <h2>
    GiftExpertApp
  </h2>
  <AddCategory
    setCategories={[Function]}
  />
  <hr />
  <ol />
</Fragment>
`;

exports[`Prubas de GifExpertApp Debe mostrar una lista de categorias 1`] = `
<Fragment>
  <h2>
    GiftExpertApp
  </h2>
  <AddCategory
    setCategories={[Function]}
  />
  <hr />
  <ol>
    <GifGrid
      category="1"
      key="1"
    />
    <GifGrid
      category="2"
      key="2"
    />
  </ol>
</Fragment>
`;
